lib/src/main/java/com/diffplug/spotless/npm/NpmProcess.java

```diff
@@ -73,8 +73,14 @@ private Process npm(String... args) {
 
 	private List<String> processCommand(String... args) {
 		List<String> command = new ArrayList<>(args.length + 1);
+		File nodeExecutable = this.npmExecutable.getParentFile().toPath().resolve("node").toFile();
+		if (!(nodeExecutable.exists() && nodeExecutable.canExecute())) {
+			throw new IllegalStateException("node must exist in the same directory as npm and be executable. Using npm @ " + this.npmExecutable.getAbsolutePath());
+		}
+		command.add(nodeExecutable.getAbsolutePath());
```

Comment:
Thank you for this PR. I'm not sure we can keep the code like this. Two reasons:
1. In #1185 you describe, that you want to use a node installation dynamically retrieved via node-gradle-plugin. With that plugin specifically it is possible, to download `npm` and `node` to different locations:
```gradle
node {
    download = true
    version = '18.13.0'
    npmVersion = '8.19.2'
    workDir = file("${buildDir}/nodejs")
    npmWorkDir = file("${buildDir}/npm")
}
```
This change would prevent such an installation from ever working with spotless prettier.
2. I think on windows, the node binary is called `node.cmd`, so this test would probably fail there.

I would suggest to change your PR to detect when the npm process fails due to missing node binary and give the user a very specific error message for this.

Regarding being able to better integrate node-gradle-plugin: I'm currently looking into allowing the user to specify PATH and/or the location of the node binary (triggered by #1381).

Reply:
Can update the error message but the problem still remains. NPM can't run without node on the $PATH and that is not desirable. Would you accept a PR that allows node to be configured the same way as NPM? (I.e a NodePathResolver?) Or even simpler, allow configuration that allows a list of paths to be appended to the path var that we can configure in the ProcessBuilder?

Reply:
> Can update the error message but the problem still remains.
Maybe you misinterpreted what I meant.

I thought of something  along the way of
```java
try {
    // npm install is triggered
} catch (Exception e) {
   if (e.getMessage().contains("failed with exit code: 127")) {
      throw new IllegalStateException("Node could not be found on the path, npm is unable to execute.");
  }
}
```

But with the changes I'm currently doing for #1381, that change would become obsolete. With that said, are you OK if we close this PR?

For the second part of your reply: 
I agree with you: we have an issue about config-phase / execution phase when interoperating spotless and gradle-node-plugin. I suggest we raise a new issue for that topic and move discussion there.